dumper/lib-ext-hive-metastore/build.gradle

```diff
@@ -78,15 +74,30 @@ shadowJar {
 
 }
 
-generateThriftSource {
-    generators {
-        java {
-            out file('build/generated-sources/thrift/main')
-            // option 'java5'
-        }
+task cleanThrift(type: Delete) {
+    delete project.projectDir.toPath().resolve('src/main/gen-java')
+}
+
+task generateThrift {
+    dependsOn cleanThrift
+}
+
+task thrift {
+    dependsOn cleanThrift, generateThrift
+}
```

Comment:
Suggestion: Gradle really likes being idempotent, so the 'delete-then-rebuild' strategy we're used to from make doesn't really work in Gradle. The trick is to use unions and syncs.

So consider a pattern where each Thrift task writes into a separate src directory.

It might be simpler to write a single thrift task which generates all thrift files into a single directory.

To avoid the Delete/ task, generate the thrift java into a directory in project.buildDir and then Sync it into src/ - the sync is a way of doing a delete+union-copy while still being idempotent.

Reply:
I've done the changes as discussed.

Reply:
Beautiful.